### ISSUE TITLE
update-lxcs.sh exiting after the update of containers

### ISSUE THREAD

**Issue by Mincka** [tteck/Proxmox#3432]
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

Running `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/update-lxcs.sh)"`.
Not skipping any container, all the containers are up and running.
The script seems to exit randomly, sometimes there are updates and it continues to the next container, sometimes, there's no update and it exits.

One run, the script is exiting after the upgrade of tailscale.
```
[Info] Updating 104 : tailscale - Boot Disk: 78% full [1.7G/2.4G used, 0.5G free]

Listing... Done
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@pve:~# 
```

In that case, after adguard, but the next time it could be the same or another one.
```
[Info] Updating 100 : adguard - Boot Disk: 50% full [1.3G/2.9G used, 1.4G free]

Listing... Done
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@pve:~# 
```

I saw that the script was modified [one month ago](https://github.com/tteck/Proxmox/commit/7d142d593e7c748b8507db9c681302a7ecdd9f06).

I've tested the [previous version](https://github.com/tteck/Proxmox/blob/ecb4294a184812b5b68f5d58fbaf65c7aad0bc4a/misc/update-lxcs.sh) and I can confirm it always works fine with the previous version, the bug is only present in the latest version.

Nothing suspicious when running the code in the shell of the LXC:
```
Adguard LXC provided by https://tteck.github.io/Proxmox/

Last login: Thu Mar 21 18:34:48 UTC 2024 on tty1
root@adguard:~# apt-get update 2>/dev/null | grep 'packages.*upgraded'; apt list --upgradable && apt-get -yq dist-upgrade 2>&1; rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED
Listing... Done
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@adguard:~# rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED
root@adguard:~# echo $?
0
root@adguard:~# 
```

### What settings are you currently utilizing?

Default Settings

### Which Linux distribution are you employing?

Debian 12

**Comment by Mellowlynx** [comment]
Works fine here (tried it on two nodes), ran it twice on one to see if it only happens when there are no updates, but no, all fine.
![image](https://github.com/user-attachments/assets/0f48ba3d-26db-410b-9898-dd1c37cb0bee)

**Comment by tteck** [comment]
Are you using SSH or directly in the Proxmox shell?

**Comment by Mincka** [comment]
> Are you using SSH or directly in the Proxmox shell?

Tested with both, same result.

**Comment by tteck** [comment]
I see no reason why adding the command `rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED` would cause an issue.

**Comment by tteck** [comment]
🔁 If I cannot reproduce the problem on my side, I cannot address an issue that appears to be functioning correctly here.

**Comment by Mincka** [comment]
```
[Info] Updating 100 : adguard - Boot Disk: 50% full [1.4G/2.9G used, 1.4G free]

Listing... Done
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@pve:~# echo $?
129
```

Script returns with  `SIGUP` when it stops.
I tried different approaches and if the commands are ran individually, the script never fails.

Since I saw this could be related to some kind of resource exhaustion, I stopped the container that uses most of the ressources (75% of total CPU). And it seems that the update script is working in that case, even if it starts the stopped container to update it.

I know that's a strange bug, and I can't explain why it started to fail with this commit which does not change anything in the overall logic. I'll continue to use a forked version with individual commands. Thank you.
